projects/laji/src/app/shared/service/util.service.ts

```diff
@@ -66,6 +70,22 @@ export class Util {
     }, {} as Partial<T>);
   }
 
+  /**
+   * Remove all undefined and null values from object
+   *
+   * @param obj object to remove keys from
+   * @param keys array of keys that should be removed
+   */
+  public static withNonNullableKeys<T extends {[prop: string]: any}, K extends keyof T>(obj: T): WithNonNullableKeys<T, K> {
```

Comment:
Shouldn't it be `removeNullableFields` or at least `withNoNullableKeys`? This name sounds to me like you are adding extra keys rather than removing nullable keys

Reply:
another alternative: `getNonNullableKeys`

Reply:
The term `NonNullable` comes from TypeScript: [NonNullable](https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype)

Though the "keys" word usage is weird... How about `withNonNullableValues`? `getNonNullableKeys` sounds to me like it would return keys of an object.

Reply:
I pushed `withNonNullableValues` as a proposal. Updated the JS doc comment also. It was copy pasted from some other utility fn so there was an extra param.

Reply:
I still don't think I would immediately understand what `withNonNullableValues` does without reading the comment. Usually the word `with` I would associate with adding something to the result, eg. `withLatestFrom` in rxjs. Maybe `excludeNullables` or invert it `selectNonNullables`.

Sorry this is getting rather pedantic... I'll give the PR an approve.